src/shared/utilities/timeoutUtils.ts

```diff
@@ -16,16 +18,18 @@ export class Timeout {
     private readonly timeoutLength: number
     private readonly timerPromise: Promise<void>
     private timerTimeout?: NodeJS.Timeout
```

Comment:
Does removing the `?` here remove the need for all the `!` assertions later? It's always being set.

Reply:
TypeScript can't detect that the resolve/reject functions are being initialized in the constructor. I added `!` to the private variable declarations to avoid redundancy.